src/frontend/context/FritekstVedtakBegrunnelserContext.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+
+import { AxiosError } from 'axios';
+import constate from 'constate';
+import deepEqual from 'deep-equal';
+
+import { useHttp } from '@navikt/familie-http';
+import { FeltState, ok, Valideringsstatus } from '@navikt/familie-skjema';
+import { byggFeiletRessurs, Ressurs, RessursStatus } from '@navikt/familie-typer';
+
+import { Behandlingstype } from '../typer/behandling';
+import { IFagsak } from '../typer/fagsak';
+import {
+    IRestPostFritekstVedtakBegrunnelser,
+    IRestVedtakBegrunnelse,
+    VedtakBegrunnelse,
+} from '../typer/vedtak';
+import { Vedtaksperiode } from '../typer/vedtaksperiode';
+import { useFagsakRessurser } from './FagsakContext';
+import { useVedtakBegrunnelser } from './VedtakBegrunnelserContext';
+
+interface Fritekster {
+    [key: string]: FeltState<string>;
+}
+
+interface IProps {
+    vedtaksperiode: Vedtaksperiode;
+    behandlingsType: Behandlingstype;
+}
+
+const [FritekstVedtakBegrunnelserProvider, useFritekstVedtakBegrunnelser] = constate(
+    ({ vedtaksperiode, behandlingsType }: IProps) => {
+        const { fagsak, settFagsak } = useFagsakRessurser();
+        const { vedtakBegrunnelser } = useVedtakBegrunnelser();
+        const { request } = useHttp();
+
+        const [redigerbarefritekster, settRedigerbarefritekster] = useState<Fritekster>({});
+        const [fritekster, settFritekster] = useState<Fritekster>({});
+        const [idPaSistOpprettetFritekst, settIdPaSistOpprettetFritekst] = useState<number>();
+
+        const [ekspandertFritekst, settEkspandertFritekst] = useState(
```

Comment:
Denne ligger kanskje litt feil hvis det er state som sier noe om perioden er ekspandert? Tror jeg ville flyttet den til vedtakbegrunnelserContext og bare hatt en map av periodene med tilhørende boolean eller noe lignende. 

Grunnen til at det kanskje blir litt rart er at den heter fritekst, men bestemmer egentlig staten til det ekspanderbare panelet for både fritekst og multiselect.